src/docs/files-provider-deprecation.rst

```diff
@@ -0,0 +1,90 @@
+Files Provider Removal
+######################
+
+The SSSD team has announced the `removal of the files provider <https://pagure.io/fesco/issue/3107>`_ feature in recent versions of SSSD in Fedora/CentOS Stream/RHEL. This document describes how to switch to using the SSSD 'proxy' provider to replace files provider functionality.
+
+Practically, there are only two use cases that currently justify usage of the files provider feature:
+
+* smart card authentication of local users
+* session recording for local users
+
+For both cases the proxy provider is a viable substitute. Otherwise, there is no benefit in SSSD handling local users. The SSSD proxy provider is just a relay, an intermediary configuration. SSSD connects to its proxy service, and then that proxy loads the specified libraries. This allows SSSD to use some resources that it otherwise would not be able to use.
+
+Follow the steps below based on the use case you are using.
+
+Smartcard Authentication for Local Users
+****************************************
+
+First, make sure to install the ``sssd-proxy`` package.
+
+.. code-block:: console
+
+    $ dnf install sssd-proxy
+
+Configure SSSD for smart card authentication as below:
+
+.. code-block:: console
+
+    [domain/myproxy]
+    id_provider = proxy
+    proxy_lib_name = files
+    local_auth_policy = only
+
+.. note:: An explicit **proxy_pam_target** is only needed if you want Smartcard authentication be an additional method besides the method provided by the **proxy_pam_target**. This is typically not needed for password authentication because pam_unix should be present in the PAM configuration to handle this.
+
```

Comment:
Hi.

in the case where `proxy_pam_target` is set you most probably want to use `local_auth_policy = enable:smartcard` because with `only` the authentication provided by `proxy_pam`target` would be ignored. So I would suggest to either mention this in the note as well or add a second example snippet. I guess I would prefer the second.

As a general comment, I wonder if it would make sense to add an example of a possible old configuration as well to make it easier for new admins to understand what they should look for. I'm thinking about cases where "I have inherited these systems from a person which is not in the company anymore and after the last updated nothing is working anymore ....". With this it would e.g.  for local Smartcard authentication more obvious to mention that the domain name should be kept or if not the domain name in the [certmap/...] sections have to be updated as well.

bye,
Sumit

Reply:
> in the case where `proxy_pam_target` is set you most probably want to use `local_auth_policy = enable:smartcard` because with `only` the authentication provided by `proxy_pam`target` would be ignored. So I would suggest to either mention this in the note as well or add a second example snippet. I guess I would prefer the second.

I added it to the existing `note` line, to keep the document concise.
> 
> As a general comment, I wonder if it would make sense to add an example of a possible old configuration as well to make it easier for new admins to understand what they should look for. I'm thinking about cases where "I have inherited these systems from a person which is not in the company anymore and after the last updated nothing is working anymore ....". With this it would e.g. for local Smartcard authentication more obvious to mention that the domain name should be kept or if not the domain name in the [certmap/...] sections have to be updated as well.

Could you provide an example configuration? I'll add it here as you suggested.

Reply:
I added a sample old/working smart card sssd.conf configuration.